crates/control/src/role_assignment.rs

```diff
@@ -155,198 +148,119 @@ impl RoleAssignment {
 
             self.role_initialized = true;
             self.last_received_spl_striker_message = Some(cycle_start_time);
-            self.team_ball = None;
         }
 
-        let send_game_controller_return_message = self
-            .last_system_time_transmitted_game_controller_return_message
-            .is_none()
-            || cycle_start_time.duration_since(
-                self.last_system_time_transmitted_game_controller_return_message
-                    .unwrap(),
-            )? > context.spl_network.game_controller_return_message_interval;
+        self.try_sending_game_controller_return_message(&context, ground_to_field)?;
 
-        let mut send_spl_striker_message = self.last_transmitted_spl_striker_message.is_none()
-            || cycle_start_time
-                .duration_since(self.last_transmitted_spl_striker_message.unwrap())?
-                > context.spl_network.spl_striker_message_send_interval;
+        // TODO: reimplement whatever this did
+        let is_in_penalty_kick = matches!(
+            context.filtered_game_controller_state,
+            Some(FilteredGameControllerState {
+                sub_state: Some(SubState::PenaltyKick),
+                ..
+            })
+        );
 
         let spl_striker_message_timeout = match self.last_received_spl_striker_message {
-            None => false,
+            None => true,
             Some(last_received_spl_striker_message) => {
                 cycle_start_time.duration_since(last_received_spl_striker_message)?
                     > context.spl_network.spl_striker_message_receive_timeout
```

Comment:
This is a horrible name... I'm trying to wrap my head around what it actually is for minutes now. If I'd only stumble upon that in the parameters, I'd have even less of an idea what this controls

Reply:
This is something like the duration we believe in the last information a striker sent us before we assume the robot which sent us the information is dead and we have to fallback and continue somehow, is it?

Reply:
so it is a `duration_this_robot_believes_in_the_last_information_of_another_striker_before_assuming_it_fell_dead_and_we_have_to_continue_doing_something_else`
@h3ndrk would be happy :laughing: